crates/llms/src/openai/mod.rs

```diff
@@ -87,13 +87,19 @@ impl Openai {
             model,
         }
     }
+
+    /// Returns true if the `OpenAI` compatible model supports [structured outputs](https://platform.openai.com/docs/guides/structured-outputs/).
+    /// This is only supported for GPT-4o models from `OpenAI` (i.e not any other compatible servers).
+    fn supports_structured_output(&self) -> bool {
+        self.client.config().api_base() == OPENAI_API_BASE && self.model.starts_with("gpt-4o")
```

Comment:
It seems like this should come from the model parameters, not be hard coded. It means that we can't be flexible to support new models without recompiling.

Reply:
This is true. I contemplated using /v1/models to get this, but it doesn't have the details. Right now, only Openai and no other providers supports structured output. 

I would really like users to be unaware of these details.